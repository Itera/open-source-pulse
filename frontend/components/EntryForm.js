// @flow
import React from 'react';
import {map} from 'lodash/fp';
import {graphql} from 'react-apollo';
import gql from 'graphql-tag';

const opts = [
  {
    key: 'ISSUE',
    description: 'A github issue you wish to fix'
  },
  {
    key: 'REPO',
    description: 'The project repo you wish to send a request to (if no issue exists)'
  },
  {
    key: 'PR',
    description: 'Your pull request'
  },
];

class EntryForm extends React.Component {
  state = {};

  onChange = (event) => {
    this.setState({[event.target.name]: event.target.value});
  }

  renderOptions = map((option) => <option key={option.key} value={option.key}>{option.description}</option>);

  render() {
    return (
      <form>
        <select name="type" onChange={this.onChange} >
          <option>Please select:</option>
          {this.renderOptions(opts)}
        </select>
        <input type="text" placeholder="URL" name="url" onChange={this.onChange} />
        <input type="button" onClick={() => this.props.onSubmit(this.state)} value="Update"/>
      </form>
    )
  }
}

const feedItemMutation = gql`
  mutation createFeedItem($feedItem: FeedItemInput!) {
    createFeedItem(feedItem: $feedItem) {
      type,
      url
    }
  }
`;

export default graphql(feedItemMutation, {
  props: ({mutate}) => ({
    onSubmit: (feedItem) => mutate({variables: {feedItem}}),
  }),
})(EntryForm);
