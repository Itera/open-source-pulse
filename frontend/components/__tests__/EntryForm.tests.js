/* eslint-env jest */
import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';

import { EntryForm } from '../EntryForm';

describe('EntryForm', () => {
  it('should render correctly', () => {
    const wrapper = mount(<EntryForm />);

    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  it('should call props.onSubmit when button is clicked', () => {
    const onSubmit = jest.fn();
    const wrapper = mount(<EntryForm onSubmit={onSubmit} />);

    wrapper.setState({ type: 'CREATED_ISSUE', url: 'http://example.com' });
    wrapper.find('button').simulate('click');

    expect(onSubmit).toBeCalledWith({ type: 'CREATED_ISSUE', url: 'http://example.com' });
  });

  it('should not call props.onSubmit when data is not filled', () => {
    const onSubmit = jest.fn();
    const wrapper = mount(<EntryForm onSubmit={onSubmit} />);

    wrapper.setState({ type: '', url: '' });
    wrapper.find('button').simulate('click');

    expect(onSubmit).not.toBeCalled();
  });
});
