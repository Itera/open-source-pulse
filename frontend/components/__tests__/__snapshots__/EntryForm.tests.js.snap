exports[`EntryForm should render correctly 1`] = `
<EntryForm>
  <styled.section>
    <section
      className="dZUSMd">
      <h2>
        Add an entry
      </h2>
      <Radio
        checked={false}
        id="radio-WORKING_ON_ISSUE"
        label="A github issue you wish to fix"
        name="type"
        onChange={[Function]}
        value="WORKING_ON_ISSUE">
        <styled.label
          htmlFor="radio-WORKING_ON_ISSUE">
          <label
            className="hnLhiS"
            htmlFor="radio-WORKING_ON_ISSUE">
            <styled.input
              checked={false}
              id="radio-WORKING_ON_ISSUE"
              name="type"
              onChange={[Function]}
              type="radio"
              value="WORKING_ON_ISSUE">
              <input
                checked={false}
                className="dKarGh"
                id="radio-WORKING_ON_ISSUE"
                name="type"
                onChange={[Function]}
                type="radio"
                value="WORKING_ON_ISSUE" />
            </styled.input>
            <span>
              A github issue you wish to fix
            </span>
          </label>
        </styled.label>
      </Radio>
      <Radio
        checked={false}
        id="radio-CREATED_ISSUE"
        label="A github issue you have created"
        name="type"
        onChange={[Function]}
        value="CREATED_ISSUE">
        <styled.label
          htmlFor="radio-CREATED_ISSUE">
          <label
            className="hnLhiS"
            htmlFor="radio-CREATED_ISSUE">
            <styled.input
              checked={false}
              id="radio-CREATED_ISSUE"
              name="type"
              onChange={[Function]}
              type="radio"
              value="CREATED_ISSUE">
              <input
                checked={false}
                className="dKarGh"
                id="radio-CREATED_ISSUE"
                name="type"
                onChange={[Function]}
                type="radio"
                value="CREATED_ISSUE" />
            </styled.input>
            <span>
              A github issue you have created
            </span>
          </label>
        </styled.label>
      </Radio>
      <Radio
        checked={false}
        id="radio-CODING_REPOSITORIY"
        label="The project repo you wish to send a request to (if no issue exists)"
        name="type"
        onChange={[Function]}
        value="CODING_REPOSITORIY">
        <styled.label
          htmlFor="radio-CODING_REPOSITORIY">
          <label
            className="hnLhiS"
            htmlFor="radio-CODING_REPOSITORIY">
            <styled.input
              checked={false}
              id="radio-CODING_REPOSITORIY"
              name="type"
              onChange={[Function]}
              type="radio"
              value="CODING_REPOSITORIY">
              <input
                checked={false}
                className="dKarGh"
                id="radio-CODING_REPOSITORIY"
                name="type"
                onChange={[Function]}
                type="radio"
                value="CODING_REPOSITORIY" />
            </styled.input>
            <span>
              The project repo you wish to send a request to (if no issue exists)
            </span>
          </label>
        </styled.label>
      </Radio>
      <Radio
        checked={false}
        id="radio-TRIAGING_REPOSITORY"
        label="Triaging issues on repo"
        name="type"
        onChange={[Function]}
        value="TRIAGING_REPOSITORY">
        <styled.label
          htmlFor="radio-TRIAGING_REPOSITORY">
          <label
            className="hnLhiS"
            htmlFor="radio-TRIAGING_REPOSITORY">
            <styled.input
              checked={false}
              id="radio-TRIAGING_REPOSITORY"
              name="type"
              onChange={[Function]}
              type="radio"
              value="TRIAGING_REPOSITORY">
              <input
                checked={false}
                className="dKarGh"
                id="radio-TRIAGING_REPOSITORY"
                name="type"
                onChange={[Function]}
                type="radio"
                value="TRIAGING_REPOSITORY" />
            </styled.input>
            <span>
              Triaging issues on repo
            </span>
          </label>
        </styled.label>
      </Radio>
      <Radio
        checked={false}
        id="radio-CREATED_PR"
        label="Your pull request"
        name="type"
        onChange={[Function]}
        value="CREATED_PR">
        <styled.label
          htmlFor="radio-CREATED_PR">
          <label
            className="hnLhiS"
            htmlFor="radio-CREATED_PR">
            <styled.input
              checked={false}
              id="radio-CREATED_PR"
              name="type"
              onChange={[Function]}
              type="radio"
              value="CREATED_PR">
              <input
                checked={false}
                className="dKarGh"
                id="radio-CREATED_PR"
                name="type"
                onChange={[Function]}
                type="radio"
                value="CREATED_PR" />
            </styled.input>
            <span>
              Your pull request
            </span>
          </label>
        </styled.label>
      </Radio>
      <Input
        label="URL"
        name="url"
        onChange={[Function]}
        placeholder="Github url"
        type="text"
        value="">
        <styled.label
          htmlFor={undefined}>
          <label
            className="etjhIf"
            htmlFor={undefined}>
            <span>
              URL
            </span>
            <styled.input
              id="url"
              name="url"
              onChange={[Function]}
              placeholder="Github url"
              type="text"
              value="">
              <input
                className="dKarGh"
                id="url"
                name="url"
                onChange={[Function]}
                placeholder="Github url"
                type="text"
                value="" />
            </styled.input>
          </label>
        </styled.label>
      </Input>
      <styled.button
        onClick={[Function]}>
        <button
          className="flabqw"
          onClick={[Function]}>
          Update
        </button>
      </styled.button>
    </section>
  </styled.section>
</EntryForm>
`;
